### PR TITLE
Signal Extensions - Behavior to define structured signal extensions

### PR DESCRIPTION
**Pull request by mikehostetler**
# Pull Request

## Description

Extend `Jido.Signal` with the capability to define behaviors that can extend this struct per the CloudEvents spec under an `extensions` key. Organize this into a Behaviour to validate extension data etc

Ensured that features such as serialization still work etc

Also added a developer syntax helper for `Jido.Signal.new/3`

## Type of Change

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [X] ✨ New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📚 Documentation update
- [ ] 🔧 Refactoring (no functional changes)
- [ ] ⚡ Performance improvement
- [ ] 🧪 Test improvement
- [ ] 🔨 Build/CI related changes

## Breaking Changes

If this PR introduces breaking changes, describe them here and provide migration instructions:

```elixir
# Before
old_function(arg)

# After  
new_function(arg, new_required_param)
```

## Testing Approach

Describe how you tested these changes:

- [ ] Added new tests
- [ ] Updated existing tests
- [ ] Manual testing performed
- [ ] Integration tests verified

## Quality Checklist

- [ ] Tests pass (`mix test`)
- [ ] Quality checks pass (`mix quality` or `mix q`)
  - [ ] Code formatted (`mix format`)
  - [ ] No compilation warnings (`mix compile --warnings-as-errors`)
  - [ ] Type checking passes (`mix dialyzer`)
  - [ ] Static analysis passes (`mix credo --all`)
- [ ] Documentation updated if needed
- [ ] Semantic commit messages used (see format below)
- [ ] Breaking changes noted above (if any)
- [ ] Related issues referenced

## Semantic Commit Message Format

This project follows conventional commit format. Ensure your commit messages follow this pattern:

```
type(scope): description

[optional body]

[optional footer(s)]
```

**Types:** `feat`, `fix`, `docs`, `style`, `refactor`, `perf`, `test`, `build`, `ci`, `chore`

**Examples:**
- `feat(bus): add new subscription filtering options`
- `fix(router): resolve pattern matching edge case`
- `docs(signal): update README with new examples`

## Related Issues

Closes #[issue number]
Relates to #[issue number]

## Additional Context

Add any other context, screenshots, or relevant information about the pull request here.

---

*Please review the [Contributing Guidelines](./CONTRIBUTING.md) before submitting your PR.*

### CODE REVIEW

**Review by ccarvalho-eng — COMMENTED** [review]
Leaving a high-level summary. Concerns are mostly around backward-compatibility (not an issue if we're not aiming to keep it backward compatible) and performance:

1. **Backward compatibility**

   * New signal extension fields risk breaking existing consumers. No defaulting / migration path visible.

2. **Behavior coupling**

   * Core now depends on extension behaviors at compile-time. Any callback change forces a full recompile of the base app. Risk of brittle coupling.

3. **Serialization fragility**

   * Extensions introduce new fields, but handling of unknown/missing ones isn’t clear. Old clients may crash or silently drop data. Needs robust fallback.

4. **Missing tests**

   * No roundtrip serialization tests with extensions. No tests for invalid/malformed extension data. Coverage gap makes regressions certain.

5. **Schema bypass**

   * Extension payloads appear to skip schema validation. This opens the door for arbitrary or malformed data to leak through.

6. **Performance risk**

   * Extension callbacks may run per signal. If routing has to match on extensions, throughput can degrade badly at scale. No caching or guard rails in sight.

7. **Concurrency hazards**

   * If extension logic manages state or spawns unsupervised tasks, it breaks OTP isolation. Could leak processes or crash cascades.

8. **Failure isolation**

   * No visible try/catch or supervisor wrapping around extension callbacks. A crash inside an extension may kill the bus/subscriber. Needs containment.

9. **Versioning gaps**

   * Extension behaviors have no versioning or compatibility strategy. Any future changes to callback contracts will break users.

File: lib/jido_signal.ex [hunk 1]
```diff
@@ -148,9 +148,18 @@ defmodule Jido.Signal do
...
   alias Jido.Signal.ID
+  alias Jido.Signal.Serialization.Serializer
   alias Jido.Signal.Serialization.TypeProvider
 
+  require Logger
+
+  @core_attrs ~w[
+    specversion id source type subject time
+    datacontenttype dataschema data jido_dispatch extensions
+  ]a
```
Comment (ccarvalho-eng) [inline lib/jido_signal.ex:161]: I didn't know we could use brackets 
I usually do something like this:
```elixir
@core_attrs ~w(
  specversion
  id
  source
  type
  subject
  time
  datacontenttype
  dataschema
  data
  jido_dispatch
  extensions
 )a
 ```
 definitely not a huge deal

**Review by ccarvalho-eng — APPROVED** [review]
This looks good to go as-is. There’s room for more cleanup and risk reduction, but we might be over-optimizing at this stage. I tried to unsee the nested cases… they’re like rabbits—multiplying every time I blink—and this PR is already big enough 😂
